src/core/server/execution_context/execution_context_config.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { TypeOf, schema } from '@kbn/config-schema';
+import { ServiceConfigDescriptor } from '../internal_types';
+
+const configSchema = schema.object({
+  enabled: schema.boolean({ defaultValue: true }),
+});
+
+/**
+ * @internal
+ */
+export type ExecutionContextConfigType = TypeOf<typeof configSchema>;
+
+export const config: ServiceConfigDescriptor<ExecutionContextConfigType> = {
+  path: 'execution_context',
+  schema: configSchema,
```

Comment:
Note: I didn't pass the config value on the client. I don't see a lot of benefits of making `ExecutionContextContainer`methods no-ops as they don't add a lot of overhead. Any objections?

Reply:
I think it's fine to have the client always be 'enabled' regardless of the config value.